hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/RatisOverReplicationHandler.java

```diff
@@ -279,16 +283,16 @@ private int createCommands(
         excess--;
       }
     }
-    replicas.removeAll(replicasRemoved);
+    eligibleAndReserveReplicas.removeFromEligible(replicasRemoved);
 
     /*
     Remove excess replicas if that does not make the container mis replicated.
     If the container was already mis replicated, then remove replicas if that
     does not change the placement count.
      */
-    Set<ContainerReplica> replicaSet = new HashSet<>(replicas);
+    Set<ContainerReplica> replicaSet = eligibleAndReserveReplicas.getReplicaSet();
```

Comment:
Nit: Maybe can add a comment on why we use the eligible + reserved replica set instead of only the eligible replica set?